R/swagger.R

```diff
@@ -4,15 +4,34 @@
 swaggerTypeInfo <- list()
 plumberToSwaggerTypeMap <- list()
 defaultSwaggerType <- "string"
+attr(defaultSwaggerType, "default") <- TRUE
+defaultSwaggerIsArray <- FALSE
+attr(defaultSwaggerIsArray, "default") <- TRUE
 
 local({
-  addSwaggerInfo <- function(swaggerType, plumberTypes, regex, converter) {
+  addSwaggerInfo <- function(swaggerType, plumberTypes,
+                             regex = NULL, converter = NULL,
+                             format = NULL,
+                             location = NULL,
+                             realType = NULL,
+                             arraySupport = FALSE) {
     swaggerTypeInfo[[swaggerType]] <<-
       list(
         regex = regex,
-        converter = converter
+        converter = converter,
+        format = format,
+        location = location,
+        arraySupport = arraySupport,
+        realType = realType
       )
 
+    if (arraySupport == TRUE) {
+      swaggerTypeInfo[[swaggerType]] <<- modifyList(
+        swaggerTypeInfo[[swaggerType]],
+        list(regexArray = paste0("(?:(?:", regex, "),?)+"),
+             converterArray = function(x) {converter(stri_split_fixed(x, ",")[[1]])})
```

Comment:
```suggestion
             # Q: Do we need to safe guard against special characters, such as `,`?
             # https://github.com/rstudio/plumber/pull/532#discussion_r439584727
             # A: https://swagger.io/docs/specification/serialization/
             # > Additionally, the allowReserved keyword specifies whether the reserved 
             # > characters :/?#[]@!$&'()*+,;= in parameter values are allowed to be sent as they are, 
             # > or should be percent-encoded. By default, allowReserved is false, and reserved characters 
             # > are percent-encoded. For example, / is encoded as %2F (or %2f), so that the parameter 
             # > value quotes/h2g2.txt will be sent as quotes%2Fh2g2.txt
             converterArray = function(x) {converter(stri_split_fixed(x, ",")[[1]])})
```